File: problems/1539-kth-missing-positive-number/solution.java
class Solution {
    public int findKthPositive(int[] arr, int k) {
        int missing = 0;
        int expected = 1;
        int i = 0;
        while (missing < k) {
            if (i < arr.length && arr[i] == expected) {
                i++;
            } else {
                missing++;
            }
            if (missing < k) {
                expected++;
            }
        }
        return expected;
    }
}

File: problems/1539-kth-missing-positive-number/solution.py
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        missing = 0
        expected = 1
        i = 0
        while missing < k:
            if i < len(arr) and arr[i] == expected:
                i += 1
            else:
                missing += 1
            if missing < k:
                expected += 1
        return expected